src/app.js

```diff
@@ -335,8 +335,7 @@ class SAFEApp extends EventEmitter {
   * Must be invoked when the client decides to connect back after the connection was lost.
   */
   reconnect() {
-    return lib.app_reconnect(this.connection)
-      .then(() => this._networkStateUpdated(null, consts.NET_STATE_CONNECTED));
+    return lib.app_reconnect(this);
```

Comment:
Isn't much simpler to just call `this._networkStateUpdated(null, consts.NET_STATE_CONNECTED);` from here, the `app_reconnect` function doesn't do anything else really.

Reply:
Calling `_networkStateUpdated` here seemed out of place because because it's used [here](https://github.com/maidsafe/safe_app_nodejs/blob/master/src/native/_app.js#L48) and [here](https://github.com/maidsafe/safe_app_nodejs/blob/master/src/native/_app.js#L71) in `src/native/_app`.